DynamicTablesPkg/Library/Acpi/X64/AcpiWsmtLib/WsmtGenerator.c

```diff
@@ -0,0 +1,262 @@
+/** @file
+  WSMT Table Generator Implementation.
+
+  This file implements the WSMT Table Generator.
+  The WSMT table is used to specify the security mitigation
+  that are enabled in the Windows OS.
+
+  Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <AcpiTableGenerator.h>
+#include <ConfigurationManagerHelper.h>
+#include <ConfigurationManagerObject.h>
+#include <IndustryStandard/WindowsSmmSecurityMitigationTable.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/TableHelperLib.h>
+#include <Protocol/AcpiTable.h>
+#include <Protocol/ConfigurationManagerProtocol.h>
+#include <X64NameSpaceObjects.h>
+
+/** This macro expands to a function that retrieves the
+    WSMT protection flags information from the Configuration Manager.
+*/
+GET_OBJECT_LIST (
+  EObjNameSpaceX64,
+  EX64ObjWsmtFlagsInfo,
+  CM_X64_WSMT_FLAGS_INFO
+  );
+
+/** The ACPI WSMT Table.
+*/
+STATIC
+EFI_ACPI_WSMT_TABLE  AcpiWsmt = {
+  ACPI_HEADER (
+    EFI_ACPI_WINDOWS_SMM_SECURITY_MITIGATION_TABLE_SIGNATURE,
+    EFI_ACPI_WSMT_TABLE,
+    EFI_WSMT_TABLE_REVISION
+    ),
+  // ProtectionFlags
+  0
+};
+
+/** Update the protection flags information in the WSMT Table.
+
+  @param [in]  CfgMgrProtocol Pointer to the Configuration Manager
+                              Protocol Interface.
+
+  @retval EFI_SUCCESS           Success.
+  @retval EFI_INVALID_PARAMETER A parameter is invalid.
+  @retval EFI_NOT_FOUND         The required object was not found.
+  @retval EFI_BAD_BUFFER_SIZE   The size returned by the Configuration
+                                Manager is less than the Object size for the
+                                requested object.
+**/
+STATIC
+EFI_STATUS
+EFIAPI
+WsmtAddProtectionFlagsInfo (
+  IN  CONST EDKII_CONFIGURATION_MANAGER_PROTOCOL  *CONST  CfgMgrProtocol
+  )
+{
+  EFI_STATUS              Status;
+  CM_X64_WSMT_FLAGS_INFO  *WsmtFlagInfo;
+
+  ASSERT (CfgMgrProtocol != NULL);
+
+  // Get the WSMT protection flag from the Platform Configuration Manager
+  Status = GetEX64ObjWsmtFlagsInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &WsmtFlagInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: WSMT: Failed to get WSMT protection flag information." \
+      " Status = %r\n",
+      Status
+      ));
+    return Status;
+  }
+
+  DEBUG ((
+    DEBUG_INFO,
+    "WSMT: Protection flags = 0x%x\n",
+    WsmtFlagInfo->ProtectionFlags
+    ));
+
+  AcpiWsmt.ProtectionFlags = WsmtFlagInfo->ProtectionFlags;

```

Comment:
Hello Abdul,
`COMM_BUFFER_NESTED_PTR_PROTECTION – Firmware setting this bit must also set the FIXED_COMM_BUFFERS bit.`

Is it possible to add a check for this ? With that, all good for me